galley/pkg/config/analysis/analyzers/auth/mtls.go

```diff
@@ -0,0 +1,185 @@
+package auth
+
+import (
+	"istio.io/api/authentication/v1alpha1"
+	"istio.io/istio/galley/pkg/config/analysis/analyzers/util"
+
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+
+	"istio.io/api/networking/v1alpha3"
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/galley/pkg/config/analysis/analyzers/auth/mtls"
+	"istio.io/istio/galley/pkg/config/meta/metadata"
+	"istio.io/istio/galley/pkg/config/meta/schema/collection"
+	"istio.io/istio/galley/pkg/config/resource"
+)
+
+// MTLSAnalyzer checks for misconfigurations of MTLS policy. More specifically,
+// it detects situations where a DestinationRule's MTLS usage is in conflict
+// with mTLS specified by a policy.
+//
+// The most common situations that this detects are: 1. A MeshPolicy exists that
+// requires mTLS, but no global destination rule
+//    says to use mTLS.
+// 2. mTLS is used throughout the mesh, but a DestinationRule is added that
+//    doesn't specify mTLS (usually because it was forgotten).
+//
+// The analyzer tries to act more generally by imagining service-to-service
+// traffic and detecting whether or not there's a conflict with regards to mTLS
+// policy. This means it will also detect explicit misconfigurations as well.
+//
+// Note this is very similar to `istioctl authn tls-check`; however this
+// inspection is all done via analyzing configuration rather than requiring a
+// connection to pilot.
+type MTLSAnalyzer struct{}
+
+// Compile-time check that this Analyzer correctly implements the interface
+var _ analysis.Analyzer = &MTLSAnalyzer{}
+
+// Metadata implements Analyzer
+func (s *MTLSAnalyzer) Metadata() analysis.Metadata {
+	return analysis.Metadata{
+		Name: "auth.MTLSAnalyzer",
+
+		// Each analyzer should register the collections that it needs to use as input.
+		Inputs: collection.Names{
+			metadata.K8SCoreV1Namespaces,
+			metadata.IstioAuthenticationV1Alpha1Meshpolicies,
+			metadata.IstioAuthenticationV1Alpha1Policies,
+			metadata.IstioNetworkingV1Alpha3Destinationrules,
```

Comment:
@sushicw, is the MeshConfig available to analyzers? We may need access to check for auto-mtls (see https://github.com/istio/api/blob/62bf0afefa2f323e70ee21b45afed3748922f8f9/mesh/v1alpha1/config.proto#L232).

Reply:
@ayj we need MeshConfig in general for other configurations like figuring out outbound policy, in general having access to it (and newer forms of MeshConfig) is going to be very helpful.

Reply:
@ayj @nrjpoddar Yes, this is something we need. Analyzers don't currently have access to this. Created https://github.com/istio/istio/issues/18931 to track adding this functionality.

The main thing is that we need to plumb the control plane namespace to the analyzers. From there, getting the right configmap and extracting values like enableAutoMtls should be straightforward.